lib/ansible/modules/cloud/amazon/aws_api_gateway.py

```diff
@@ -271,9 +320,31 @@ def configure_api(client, api_data=None, api_id=None, mode="overwrite"):
 
 
 @AWSRetry.backoff(**retry_params)
-def create_deployment(client, api_id=None, stage=None, description=None):
-    # we can also get None as an argument so we don't do this as a default
-    return client.create_deployment(restApiId=api_id, stageName=stage, description=description)
+def create_deployment(client, api_id=None, stage=None, description=None, cache_enabled=False, cache_size='0.5',
+        variables=None, canary_settings=None, tracing_enabled=False):
```

Comment:
This method should probably be refactored to use keyword arguments instead, to reduce params list here.